dev/bench/odbc/main.cc

```diff
@@ -0,0 +1,186 @@
+#include <sql.h>
+#include <sqlext.h>
+#include <sqltypes.h>
+#include <stdio.h>
+#include <string.h>
+#include <chrono>
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <string>
+#include <vector>
+
+#define ERRMSG_LEN 200
+
+SQLINTEGER checkError(SQLRETURN rc, SQLSMALLINT handleType, SQLHANDLE handle,
+                      SQLWCHAR* errmsg) {
+  SQLRETURN retcode = SQL_SUCCESS;
+
+  SQLSMALLINT errNum = 1;
+  SQLWCHAR sqlState[10];
+  SQLINTEGER nativeError;
+  SQLWCHAR errMsg[ERRMSG_LEN];
+  SQLSMALLINT textLengthPtr;
+
+  if ((rc != SQL_SUCCESS) && (rc != SQL_SUCCESS_WITH_INFO) && (rc != SQL_NO_DATA)) {
+    SQLLEN numRecs = 0;
+    SQLGetDiagField(SQL_HANDLE_STMT, handle, 0, SQL_DIAG_NUMBER, &numRecs, 0, 0);
+    while (retcode != SQL_NO_DATA) {
+      retcode = SQLGetDiagRecW(handleType, handle, errNum, sqlState, &nativeError, errMsg,
```

Comment:
Note that SQLWCHAR* is _not_ guaranteed to be the same as wchar_t*. It depends on both the driver manager and how it is built. unixODBC uses unsigned short (2-bytes) by default while wchar_t is 4-bytes on most *nix OSes.

It is generally easier to use ANSI functions outside of Windows because of this (ANSI functions should be UTF-8 on *nix platforms)

Reply:
Switched to using the ANSI version